reporting/report_test.go

```diff
@@ -41,3 +43,14 @@ JPcHeO7M6FohKgcEHX84koQDN98J/L7pFlSoU7WOl6f8BKavIdeSTPS9qQYWdQuT
 		t.Errorf("got %+v want %+v", key, results)
 	}
 }
+
+func TestWriteReportInNonExistingDir(t *testing.T) {
+	report := Init()
+
+	err := report.WriteFile([]string{"test_temp_dir/report"}, &config.Config{Name: "report", Version: "5"})
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
I think it will be better to create the temp file on a temp dir in the system temp folder.

Also, test it with non-exist sub-folder.

```suggestion
	tempDir := os.TempDir()
	resultFile := filepath.Join(tempDir, "test_temp_dir", "subdir", "report.yaml")
	err := report.WriteFile([]string{resultFile}, &config.Config{Name: "report", Version: "5"})
	if err != nil {
		t.Error(err)
	}
```